packages/design-system/src/components/StepList/StepLink.tsx

```diff
@@ -16,7 +16,7 @@ export interface StepLinkProps {
   screenReaderText?: string;
   className?: string;
   onClick?: (href?: string, stepId?: string) => any;
-  component?: React.ComponentType<StepLinkComponentProps> | keyof JSX.IntrinsicElements;
+  component?: React.ComponentType<StepLinkComponentProps> | 'a';
```

Comment:
This was changed to fix a build error happening with web component work. Web components are defined as intrinsic elements globally (I think this was to fix some issues with `ds-choice` and testing) and as a result, StepLink component types included all web components.

Looking at StepLink, it has an `href` prop so why does it allow for any component type when this prop indicates it should be a link? There's also an `onClick` applied to this link that shouldn't exist (`onClick` is button behavior, links use the History API afaik so this sort of manual controlling of browser history is frowned upon - afaik).

In short, this should fix it and StepLink needs future love.